app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt

```diff
@@ -123,6 +122,31 @@ class NavigationOverlayRobot {
             return SettingsRobot.Transition()
         }
 
+        /*
+         * Navigate to the settings button using keypresses and open, and maintain focus.
+         * Using click() to select buttons removes focus, so this is an alternative way to open
+         * Settings.
+         */
+        fun navigateToSettingsAndOpen(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
+            // We hard-code this navigiation pattern because making a generic way to linearly navigate
+            // is very difficult within Espresso. Espresso does not allow branching, and relies on
+            // assertions, so a generic keypress navigation with conditional checks is not worth writing
+            // for a single test. Can revisit if more focus-related tests are written.
+            device.apply {
+                // This will need to change if the button layout changes.
```

Comment:
Optional nit: += `However, such layout changes are infrequent, and updating this will be easy`